scripts/validate

```diff
@@ -5,12 +5,46 @@ cd $(dirname $0)/..
 
 ./scripts/prepare
 
+# Remove special changes from optional flags before checking if Git is dirty
+for f in packages/*; do
+    if [[ -f ${f}/package.yaml ]]; then
+        split_crds=$(cat ${f}/package.yaml | yq r - generateCRDChart.enabled)
+		if [[ "${split_crds}" == "true" && -d ${f}/charts-crd ]]; then
+			./scripts/clean-crds ${f}
+		fi
+        providesGVR="$(cat ${f}/package.yaml | yq r - providesGVR)"
+		if [[ -n ${providesGVR} && -f ${f}/charts/Chart.yaml ]]; then
+			if [[ "$(yq r ${f}/charts/Chart.yaml 'annotations[catalog.cattle.io/provides-gvr]')" == "${providesGVR}" ]]; then
+				yq d -i ${f}/charts/Chart.yaml "annotations[catalog.cattle.io/provides-gvr]" 
```

Comment:
@mrajashree I pulled out your local changes into my fork and tested this out and it seems to be passing validate on your script as expected... with one exception. I had to make a tiny change to your chart in this commit: https://github.com/aiyengar2/charts/commit/e152e5214dfbb31a6499e22f970da3c558bd03fb.

The technical reason why this change is required is because `yq` doesn't support preserving the formatting with in place writing (based on https://github.com/mikefarah/yq/issues/465), so changes like `yq d -i` or `yq w -i` sprinkled throughout our codebase essentially reformat your `Chart.yaml`'s formatting to a standard YAML format.

Specifically, this causes an issue in the `validate` script since it checks to see if your Git is dirty when performing the validate (i.e. you have changes that have not been committed to your patch or directly to your chart). Since you saved your `Chart.yaml` with a different format than what the output of `yq [d/w] -i ${f}/charts/Chart.yaml` would have generated, the validate will always fail since it will always assume your git is dirty.

TLDR, if you see an issue with the make validate, just make sure the `Chart.yaml`'s output matches the output provided by `yq r ${f}/charts/Chart.yaml`, where `f=packages/backup-restore-operator` or something similar, and run `make validate` once more.

Reply:
yup I'd noticed that too, tested my commit with this PR and the ci works fine

Reply:
@aiyengar2 If yq is going to be changing formatting of files then failing a build this needs to be more obvious, and we need to include formatting as part of the make steps a dev will run locally similar to go-fmt so anyone will be compliant.

Reply:
Easy solution is just to remove these from package.yaml altogether and just manually put in the patch file. The dual way was too confusing. Since its disconnected from the CRD generation and pushed onto parent it doesn't make sense in package.yaml anymore and will greatly simplify scripts.

Reply:
@dramich the new last commit on the PR https://github.com/rancher/charts/pull/620/commits/bf9a6228754c99af14d405566484cf5ac44cae67 ensures that yq won't break the build. @mrajashree with the last commit, you should no longer need to make the `Chart.yaml` change to get your chart to pass CI.

I tried adding formatting as part of the make validate instead, but it seems like there could be discrepancies between the output `yq r` gives on the machine that runs our tests vs. the local dev environment.

re: @cbron's comments, I agree that `providesGVR` should be removed from the package.yaml to simplify the scripts, but I think we still need to modify the Chart.yaml for `auto-install` annotations for CRD charts.

Reply:
We talked about this offline and to simplify things we are just removing all the providesGVR and `git update-index` stuff. Scripts has to stay small and simple until we rewrite. The downside is that validate might fail for original charts on chart.yaml formatting, we can make a separate make command for that later, but right now its a nice-to-have.